cql/src/main/java/org/apache/cassandra/stargate/transport/internal/Message.java

```diff
@@ -340,20 +342,20 @@ public void decode(ChannelHandlerContext ctx, Frame frame, List results) {
 
           if (connection != null
               && ((ServerConnection) connection).clientInfo() != null
-              && ((ServerConnection) connection).clientInfo().getAuthToken() != null) {
+              && ((ServerConnection) connection).clientInfo().getAuthenticatedUser() != null) {
             if (customPayload != null) {
               message
                   .getCustomPayload()
                   .put(
                       "token",
-                      StandardCharsets.UTF_8.encode(
-                          ((ServerConnection) connection).clientInfo().getAuthToken()));
+                      authenticatedUserToByteBuffer(
+                          ((ServerConnection) connection).clientInfo().getAuthenticatedUser()));
```

Comment:
Could we cache this in `ClientInfo` to avoid re-serialization on every message?

Reply:
Oh, good call.